src/gui/layout/qgslayoutmanualtablewidget.cpp

```diff
@@ -185,6 +186,13 @@ void QgsLayoutManualTableWidget::setTableContents()
   else
   {
     mEditorDialog = new QgsTableEditorDialog( this );
+    if ( QgsPrintLayout *layout = qobject_cast< QgsPrintLayout * >( mTable->layout() ) )
+    {
+      if ( layout->atlas() && layout->atlas()->coverageLayer() && layout->atlas()->enabled() )
+      {
+        mEditorDialog->setLayer( layout->atlas()->coverageLayer() );
+      }
+    }
```

Comment:
This is the correct way to retrieve the layer, which works for both atlas & reports:

```suggestion
    if ( QgsLayout * layout =mTable->layout() )
    {
      mEditorDialog->setLayer( layout->reportContext().layer() );
    }
```